### ISSUE TITLE
examples missing

### ISSUE THREAD

**Issue by psychocoderHPC** [ComputationalRadiationPhysics/cupla#120]
cupla has no examples of how atomic functions or other device functions should be used.

Please add more documentation.

**Comment by sbastrakov** [comment]
@psychocoderHPC are you thinking of a simple demonstration, like the `simpleAtomicIntrinsics` sample in CUDA? Or smth more realistic?

**Comment by psychocoderHPC** [comment]
Yes a simple example like `simpleAtomicIntrinsics` would be nice. We should also have the usage of the memory hierarchy within those examples.
